source/extensions/filters/http/stateful_session/stateful_session.cc

```diff
@@ -45,6 +45,9 @@ Http::FilterHeadersStatus StatefulSession::decodeHeaders(Http::RequestHeaderMap&
     config = route_config->statefuleSessionConfig();
   }
   session_state_ = config->createSessionState(headers);
+  if (session_state_ == nullptr) {
```

Comment:
Oops. What was the previous behavior?

Reply:
In the previous implementation, we will always create a valid session state instance and try to parse override host in the request phase and then try to update session state in the response phase. 

And in the new implementation, the session state factory can create null session state according it's config and request headers.

---

For example, the cookie based session state factory will return null session state if the request path not match the cookie path in configuration.

Because the browsers will attach cookies to request based on the request path, so if request path does not match the cookie path:
- New generated cookie by the session state will not work for this API and may break other APIs' cookie.
- Session cookie in the request is not generated by the current session state and we can safely ignore it.